### DISCUSSION TITLE
DAI 2024-2025 - Java IOs benchmarking - Nicolas Bovard

### DISCUSSION THREAD

**Discussion by nicolasbvd** [heig-vd-dai-course/heig-vd-dai-course#391] [Show and tell]
Repository: [GitHub](https://github.com/nicolasbvd/heig-vd-dai-course-java-ios-practical-content)

## Results

| File Size       | Operation         | BINARY (ms) | BUFFERED_BINARY (ms) | TEXT (ms) | BUFFERED_TEXT (ms) |
|-----------------|-------------------|-------------|----------------------|-----------|--------------------|
| 1 byte          | Write             | 153         | 152                  | 158       | 155                |
|                 | Read              | 146         | 150                  | 150       | 153                |
| 1024 bytes      | Write             | 143         | 158                  | 166       | 146                |
|                 | Read              | 161         | 160                  | 154       | 153                |
| 1 MiB (1048576 bytes) | Write       | 1358        | 202                  | 300       | 227                |
|                 | Read              | 462         | 207                  | 300       | 201                |
| 5 MiB (5242880 bytes) | Write       | 7265        | 286                  | 531       | 330                |
|                 | Read              | 1787        | 273                  | 467       | 333                |

### 1. Which type of stream is the most efficient for each use case?
   **Binary:**  
   - 1B: BINARY  
   - 1KiB: BINARY  
   - 1MiB: BUFFERED_BINARY  
   - 5MiB: BUFFERED_BINARY  

   **Text:**  
   - 1B: TEXT and BUFFERED_TEXT  
   - 1KiB: BUFFERED_TEXT  
   - 1MiB: BUFFERED_TEXT  
   - 5MiB: BUFFERED_TEXT  

   In general, the buffered binary is the quickest.

### 2. Why is it more efficient than the other types of streams?
Buffered streams are more efficient because they reduce the number of I/O operations.

### 3. What is the difference between binary data and text data?
Binary data is stored in a format readable by computer, while text data represents human characters.

### 4. What is a character encoding?
It is when we transform characters in sequence of bits.

### 5. Why is this benchmark methodology important?
We can better know when certain streams are more efficient than others.

**Comment by ludelafo** [comment]
Thank you!

You can always find the solution in the solution repository if you ever need to compare your work with the official one here: https://github.com/heig-vd-dai-course/heig-vd-dai-course-solutions.

Do not hesitate to ask questions if something is not clear, either in this discussion or in the Q&A category of the GitHub Discussions here: https://github.com/orgs/heig-vd-dai-course/discussions/categories/q-a.
